src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1799,6 +1805,33 @@ bool core::check_tx_semantic(const transaction& tx, bool keeped_by_block) const
     return true;
 }
 //-----------------------------------------------------------------------------------------------
+void core::check_service_node_ip_address() {
+    if (!m_service_node || m_has_ip_check_disabled || service_node_list.debug_allow_local_ips) {
```

Comment:
These 3 checks here ensure we're a service node, we don't have IP checking disabled, and we're not running with local IP's allowed.